scripts/check-pkg-for-release.js

```diff
@@ -269,6 +272,44 @@ function checkComponentExportedAddedInMDCPackage(ast) {
   return isExported;
 }
 
+function checkUsedDependenciesMatchDeclaredDependencies() {
+  const files = readDirRecursive(
+    PACKAGE_RELATIVE_PATH,
+    (fileName) => {
+      return fileName[0] !== '.'
+        && fileName !== 'node_modules' && fileName !== 'test';
+    }
+  );
+
+  const usedDeps = new Set();
+  const importMatcher = RegExp('(@import|from) ["\'](@material/[^/"\']+)', 'g');
+  files.forEach((file) => {
+    if (file.endsWith('.scss') || file.endsWith('.ts') && !file.endsWith('.d.ts')) {
+      const src = fs.readFileSync(path.join(PACKAGE_RELATIVE_PATH, file), 'utf8');
+      while ((dep = importMatcher.exec(src)) !== null) {
+        usedDeps.add(dep[2]);
+      }
+    }
+  });
+
+  const declaredDeps = new Set(
+    Object.keys(CLI_PACKAGE_JSON.dependencies ? CLI_PACKAGE_JSON.dependencies : [])
+      .filter((key) => key.startsWith('@material/')));
+
+  const usedButNotDeclared = [...usedDeps].filter((x) => !declaredDeps.has(x));
+  const declaredButNotUsed = [...declaredDeps].filter((x) => !usedDeps.has(x));
+
+  assert.equal(usedButNotDeclared.length, 0,
+    'FAILURE: Component ' + CLI_PACKAGE_JSON.name +
+    ' uses one or more MDC dependencies not declared in package.json.\n' +
+    getMissingDependencyRemedy(usedButNotDeclared));
+
+  assert.equal(declaredButNotUsed.length, 0,
+    'FAILURE: Component ' + CLI_PACKAGE_JSON.name +
+    ' declares one or more MDC dependencies in package.json that are unused.\n' +
```

Comment:
This would break #5441, which I think is a valid use case. While `@material/tab-indicator` is not directly used in code, but it is required in the DOM markup.

I would expect `npm install @material/tab-bar` to install all required dependencies for both code and DOM.

Thoughts?

Reply:
What do you mean by referenced in DOM?

If a subcomponent is used, clients would've to explicitly install all the packages required to render that component. I think we should update installation section of [tab-bar README](https://github.com/material-components/material-components-web/tree/master/packages/mdc-tab-bar#installation) file to include `@material/tab-indicator`, `@material/tab-scroller` & `@material/tab`

Reply:
I think [Allan just found](https://github.com/material-components/material-components-web/issues/5441#issuecomment-574772803) that tab-indicator is depended on by tab, and that does install it correctly. We may not need to do anything there or update the readme since all the subcomponents are installed alongside it.